GoogleForCreators/web-stories-wp
# Display recently used fonts

## Summary

Tracks all used fonts of the Story and displays these as first options in the list of the new font picker.


## Relevant Technical Choices

- Counts all the fonts again if a text element was added/removed or if the length of the pages changes.



## To-do



## User-facing changes

- All fonts used in the Story are displayed in the font picker and sorted in the order of the most used font.


## Testing Instructions
Note: this is currently behind a feature flag so the feature flag (`newFontPicker`) needs to be set to `true` for testing.

1. Add a few text elements with different font familys.
2. Open the font picker
3. Verify that all the used fonts are displayed among the recently used fonts.
4. Now remove all the text elements except for one.
5. Verify that the fonts were removed from the recently used fonts, except for the font used for the last remaining text element.


---



Fixes #1955
